src/IIIFPresentation/API/Features/Storage/Requests/GetHierarchicalCollection.cs

```diff
@@ -119,24 +119,37 @@ INNER JOIN
     level = max_level
   AND slug = slug_array[max_level]
   AND tree_path.customer_id = {request.CustomerId}";
+                
+        Collection? collection;
 
-        var storage = await dbContext.Collections.FromSqlRaw(query).OrderBy(i => i.CustomerId)
-            .FirstOrDefaultAsync(cancellationToken);
+        if (request.Slug.Equals(string.Empty, StringComparison.OrdinalIgnoreCase))
+        {
+            collection = await dbContext.Collections.AsNoTracking().FirstOrDefaultAsync(
+                s => s.CustomerId == request.CustomerId && s.Parent == null,
+                cancellationToken);
+        }
+        else
+        {
+            collection =  await dbContext.Collections.FromSqlRaw(query).OrderBy(i => i.CustomerId)
+                .FirstOrDefaultAsync(cancellationToken);
+        }
 
-        IQueryable<Collection>? items = null;
+        List<Collection>? items = null;
 
-        if (storage != null)
+        if (collection != null)
         {
-            items = dbContext.Collections.Where(s => s.CustomerId == request.CustomerId && s.Parent == storage.Id);
+            items = await dbContext.Collections
+                .Where(s => s.CustomerId == request.CustomerId && s.Parent == collection.Id)
+                .ToListAsync(cancellationToken: cancellationToken);
 
             foreach (var item in items)
             {
-                item.FullPath = $"{request.Slug}/{item.Slug}";
+                item.FullPath = string.IsNullOrEmpty(request.Slug) ? item.Slug : $"{request.Slug}/{item.Slug}";
```

Comment:
This is identical to what's used in `UpdateCollection` - can that go in a helper to manage paths?